hpedockerplugin/hpe/hpe_3par_common.py

```diff
@@ -939,18 +942,29 @@ def _set_flash_cache_policy_in_vvs(self, flash_cache, vvs_name):
                 exception.PluginException(ex)
 
     def _add_volume_to_volume_set(self, volume, volume_name,
-                                  cpg, flash_cache):
-        vvs_name = self._get_3par_vvs_name(volume['id'])
-        domain = self.get_domain(cpg)
-        self.client.createVolumeSet(vvs_name, domain)
-        try:
-            self._set_flash_cache_policy_in_vvs(flash_cache, vvs_name)
-            self.client.addVolumeToVolumeSet(vvs_name, volume_name)
-        except Exception as ex:
-            # Cleanup the volume set if unable to create the qos rule
-            # or flash cache policy or add the volume to the volume set
-            self.client.deleteVolumeSet(vvs_name)
-            raise exception.PluginException(ex)
+                                  cpg, flash_cache, vvs_name=None):
+        if vvs_name is not None:
+            try:
+                if flash_cache is not None:
+                    self._set_flash_cache_policy_in_vvs(flash_cache, vvs_name)
+                self.client.addVolumeToVolumeSet(vvs_name, volume_name)
+            except Exception as ex:
+                msg = _('VV Set %s does not exist.') % vvs_name
```

Comment:
We can probably get exception for different reasons here in which case logging the actual reason would be more helpful.
E.g. msg = "Failed to set flash-cache policy or add volume to VV set %s - %s" % (vvs_name, ex.message)